File: site_adapters/services/engine/scaffolds/reader.py
"""
Reader script - replaces defuddle engine.

Parameters:
  url: str - URL being processed
  config: dict - merged config (default + reader sections)

Return:
  str (HTML content) or dict with content and optional title
"""


def extract(url, config, output_path=None):
    # Access config fields
    defuddle_args = config.get("defuddle_args", {})
    headers = config.get("headers", {})

    # TODO: implement your reader logic
    # Example using BeautifulSoup:
    # from bs4 import BeautifulSoup
    # import requests
    # resp = requests.get(url, headers=headers)
    # soup = BeautifulSoup(resp.text, "html.parser")
    # article = soup.find("article")
    # return str(article) if article else ""

    return ""

File: site_adapters/services/engine/scaffolds/reader.js
/**
 * Reader script - replaces defuddle engine.
 *
 * Input (stdin JSON):
 *   url: string - URL being processed
 *   config: object - merged config (default + reader sections)
 *
 * Output (stdout):
 *   HTML string or JSON { content: string, title: string|null }
 */

const fs = require('fs');

const input = JSON.parse(fs.readFileSync('/dev/stdin', 'utf8'));
const { url, config } = input;

// Access config fields
const defuddleArgs = config.defuddle_args || {};
const headers = config.headers || {};

// TODO: implement your reader logic
// Example using Readability:
// const { JSDOM } = require('jsdom');
// const { Readability } = require('@mozilla/readability');
// const resp = await fetch(url, { headers });
// const html = await resp.text();
// const doc = new JSDOM(html, { url });
// const reader = new Readability(doc.window.document);
// const article = reader.parse();
// console.log(article.content);

console.log('');
